tests/test_sklearn_voting_classifier_converter.py

```diff
@@ -313,25 +324,10 @@ def test_voting_classifier_one_more_input(self):
                         estimators=[
                             (
                                 "est",
-                                Pipeline(
```

Comment:
Any particular reason why you changed this test?

Reply:
Reason for this is strange 'SklearnConcat' conversion function that converts all inputs to first input type and concat inputs to single tensor. 

![voting_classifier](https://github.com/onnx/sklearn-onnx/assets/52438975/64e30a5e-daf9-43b1-89e7-c242acfef985)
This image is `test_voting_classifier_one_more_input()` with old pipeline. As you can see, SklearnConcat conversion cast all non string input nodes to StringTensorType and applies Concat operator. And after that LabelEncoder will fail because keys type is int64 by Original Encoder, but input variable type is string.

Reply:
I think instead of using SklearnConcat converter you need add Concat/FeatureVectorizer operator to each converter that is needing single input in current implementation.
For example, SklearnVotingClassifier/Regressor doesn't need single input because each estimator in voting model is operator (Pipeline or Estimator) that can convert input variables to prediction/probability variables. 
And you can add Concat/FeatureVectorizer operator to, for example, SklearnRandomForest converter, SklearnLinearRegressor converter and etc.

Reply:
I see. But is the previous pipeline a valid pipeline from scikit-learn point of view? If so, we should do something about it as well.

Reply:
I found solution for problem with this test case. Now OrdinalEncoder converter set keys types for LabelEncoder by input variable dtype, not by OrdinalEncoder `categories_` field type.
I have added these changes. In addition, I found some other bugs that I fixed in the last commit. They are described in the commit message

Reply:
Thanks for fixing it.